openfe_gromacs/protocols/gromacs_md/md_settings.py

```diff
@@ -0,0 +1,668 @@
+    'yes': Generate velocities in gmx grompp according to a Maxwell
+        distribution at temperature gen-temp, with random seed gen-seed.
+        This is only meaningful with integrator=md.
+    Default 'yes'.
+    """
+    gen_temp: FloatQuantity["kelvin"] = 298.15 * unit.kelvin
+    """
+    Temperature for Maxwell distribution. Default 298.15 * unit.kelvin.
+    """
+    gen_seed: int = -1
+    """
+    Used to initialize random generator for random velocities, when gen-seed is
+    set to -1, a pseudo random seed is used. Default -1.
+    """
+
+    # # # Bonds # # #
+    constraints: Literal[
+        "none", "h-bonds", "all-bonds", "h-angles", "all-angles"
+    ] = "h-bonds"
+    """
+    Controls which bonds in the topology will be converted to rigid holonomic
+    constraints. Note that typical rigid water models do not have bonds, but
+    rather a specialized [settles] directive, so are not affected by this
+    keyword. Allowed values are:
+    'none': No bonds converted to constraints.
+    'h-bonds': Convert the bonds with H-atoms to constraints.
+    'all-bonds': Convert all bonds to constraints.
+    'h-angles': Convert all bonds to constraints and convert the angles that
+        involve H-atoms to bond-constraints.
+    'all-angles': Convert all bonds to constraints and all angles to
+         bond-constraints.
+    Default 'h-bonds'
+    """
+    constraint_algorithm: Literal["lincs", "shake"] = "lincs"
+    """
+    Chooses which solver satisfies any non-SETTLE holonomic constraints.
+    Allowed values are:
+    'lincs': LINear Constraint Solver. With domain decomposition the parallel
+        version P-LINCS is used. The accuracy in set with lincs-order, which
+        sets the number of matrices in the expansion for the matrix inversion.
+        After the matrix inversion correction the algorithm does an iterative
+        correction to compensate for lengthening due to rotation. The number of
+        such iterations can be controlled with lincs-iter. The root mean square
+        relative constraint deviation is printed to the log file every nstlog
+        steps. If a bond rotates more than lincs-warnangle in one step, a
+        warning will be printed both to the log file and to stderr.
+        LINCS should not be used with coupled angle constraints.
+    'shake': SHAKE is slightly slower and less stable than LINCS, but does work
+        with angle constraints. The relative tolerance is set with shake-tol,
+        0.0001 is a good value for “normal” MD. SHAKE does not support
+        constraints between atoms on different decomposition domains, so it can
+        only be used with domain decomposition when so-called update-groups are
+        used, which is usally the case when only bonds involving hydrogens are
+        constrained. SHAKE can not be used with energy minimization.
+    Default 'lincs'
+    """
+    continuation: Literal["no", "yes"] = "no"
```

Comment:
Should this be user facing? Continuations is more something we would control on our end.

Reply:
Will move this to be not user interfacing.